transports/websocket-websys/src/lib.rs

```diff
@@ -19,27 +19,51 @@
 // SOFTWARE.
 
 //! Libp2p websocket transports built on [web-sys](https://rustwasm.github.io/wasm-bindgen/web-sys/index.html).
+//!
+//! # Node.JS
+//!
+//! This crate will not work out of the box in a Node.JS environment, as there
+//! is no built-in WebSocket implementation, you can fix this by adding the
+//! following line to your project:
+//!
+//! ```js
+//! global.WebSocket = require('isomorphic-ws');
+//! ```
+//!
+//! This will polyfill the `WebSocket` global variable with the `isomorphic-ws`
+//! package.
 
 mod web_context;
 
-use bytes::BytesMut;
-use futures::task::AtomicWaker;
-use futures::{future::Ready, io, prelude::*};
-use js_sys::Array;
-use libp2p_core::{
-    multiaddr::{Multiaddr, Protocol},
-    transport::{ListenerId, TransportError, TransportEvent},
-};
-use send_wrapper::SendWrapper;
 use std::cmp::min;
+use std::pin::Pin;
 use std::rc::Rc;
 use std::sync::atomic::{AtomicBool, Ordering};
 use std::sync::Mutex;
-use std::{pin::Pin, task::Context, task::Poll};
-use wasm_bindgen::{prelude::*, JsCast};
+use std::task::{Context, Poll};
+
+use bytes::BytesMut;
+use futures::future::Ready;
+use futures::io;
+use futures::prelude::*;
+use futures::task::AtomicWaker;
+use js_sys::Array;
+use libp2p_core::multiaddr::{Multiaddr, Protocol};
+use libp2p_core::transport::{ListenerId, TransportError, TransportEvent};
+use send_wrapper::SendWrapper;
+use wasm_bindgen::prelude::*;
+use wasm_bindgen::JsCast;
 use web_sys::{CloseEvent, Event, MessageEvent, WebSocket};
 
-use crate::web_context::WebContext;
+use crate::web_context::{IntervalHandle, WebContext};
+
+fn ensure_websocket_supported() {
+    if !WebContext::is_websocket_supported() {
+        panic!(
+            "Websockets are not supported in this environment, you can use the `isomorphic-ws` npm package to polyfill `global.WebSocket` to enable support."
+        );
+    }
+}
```

Comment:
Are we using this more than once? If no then we can just inline it.

Reply:
fixed in [a335ef9](https://github.com/libp2p/rust-libp2p/pull/5025/commits/a335ef99ebaf8aac64286ba9b14702412f11581b)